credentialdigger/cli/hook.py

```diff
@@ -0,0 +1,180 @@
+        Command line parser
+    """
+    parser.set_defaults(func=run)
+
+
+def system(*args, **kwargs):
+    """Run a command and get the result."""
+    kwargs.setdefault('stdout', subprocess.PIPE)
+    proc = subprocess.Popen(args, **kwargs)
+    out, err = proc.communicate()
+    return out
+
+
+def print_msg(msg):
+    """Print messages to /dev/tty"""
+    subprocess.run(f'echo \"\n{msg}\n\" > /dev/tty',
+                   shell=True,
+                   stdout=subprocess.PIPE)
+
+
+def ask_commit(str_discoveries):
+    """Ask for the commit confirmation in case of possible leaks"""
+
+    msg = 'You have the following disoveries:\n' \
+          f'{str_discoveries}\ncontinue? (y/n)'
+    print_msg(msg)
+
+    sys.stdin = open('/dev/tty', 'r')
+    # Create a process on /dev/tty to capture the input (commit or not)
+    # It reads the input, saves it in userinput and echos it
+    # subprocess.check_output return the output of the command i.e., userinpput
+    user_input = subprocess.check_output('read -p \"\" userinput && echo '
+                                         '\"$userinput\"',
+                                         shell=True, stdin=sys.stdin).rstrip()
+
+    return user_input.decode('utf-8')
+
+
+def run(args):
+    """Run credential digger on staged files
+
+    Parameters
+    ----------
+    args: `argparse.Namespace`
+        Arguments from command line parser.
+
+    Returns
+    -------
+        It exits with success code 0 if there are no discoveries, otherwise
+        it exits with an error code != 0 printing the leaks
+    """
+
+    files_status = system('git', 'diff', '--name-status', '--staged'
+                          ).decode('utf-8').splitlines()
+    files = []
+    for fs in files_status:
+        stats = fs.split()
+        status = stats[0]
+        # Takes the last filename which is, in case of renamed files,
+        # the new name
+        filename = stats[-1]
+        # D = deleted files
+        # R = renamed files
+        # It considers the first char because the renamed files are displayed
+        # as Rxxx where xxx is a number
+        if status[0] not in 'DR':
+            files.append(filename)
+
+    if args.db_path:
+        db_path = args.db_path
+    else:
+        db_path = str(Path.home() / '.local' / 'data.db')
+
+    c = SqliteClient(path=db_path)
+
+    if args.rules:
+        c.add_rules_from_file(args.rules)
+    elif not c.get_rules():
+        c.add_rules_from_file('./ui/backend/rules.yml')
+
+    new_discoveries = []
+    subprocess.run(f'echo \"\nChecking files={files} \" > /dev/tty',
+                   shell=True,
+                   stdout=subprocess.PIPE)
+
+    # For optimization purposes, the PathModel and the PasswordModel are
+    # separated, otherwise scan_path will call both models for each file
+    # With this implementation the discoveries are accumulated and the
+    # PasswordModel will be used only once for the password discoveries
+    for file in files:
```

Comment:
don't use `file` since it's a reserved keyword in python. Better call it `staged_file`